fastavro/reader.py

```diff
@@ -42,12 +42,33 @@
 MASK = 0xFF
 
 
-def read_null(fo, schema):
+class SchemaResolutionError(Exception):
+    pass
+
+
+def match_types(writer_type, reader_type):
+    # The writer and reader types might not be avro primitives, they could be
+    # named records or enums
+    if isinstance(writer_type, list) or isinstance(reader_type, list):
```

Comment:
This was intended. The reason is that if the writer was an array of ints and the reader was an array of a union of ints or strings, then that should be no problem and the resolution will happen in read_union.

However, when adding some tests to show this, I found an issue. Namely, if the writer schema is a boolean or other primitive type, and the reader schema is something else, a SchemaResolutionError isn't thrown because the check isn't done.